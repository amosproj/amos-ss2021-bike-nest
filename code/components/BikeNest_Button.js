import React from 'react';
import { Text, TouchableOpacity, View, Image } from 'react-native';
import { mainStyles } from "../styles/MainStyles";

export const ButtonStyle = Object.freeze({ "big": 1, "medium": 2, "small": 3, });

export default function BikeNest_Button(props) {

    function onPress() {
        if (props.onPress != null) {
            props.onPress();
        } else {
            console.error("onPress is null");
        }
    }

    let showIcon = () => {
        if (props.iconPath != null) {
            return (
                <View style={mainStyles.buttonImage}>
                    <Image source={props.iconPath} />
                </View>
            )
        }
    }

    let setButtonStyle = () => {
        let buttonStyle = mainStyles.buttonSmall;

        if (props.type != null) {
            if (props.type === ButtonStyle.big) {
                buttonStyle = mainStyles.buttonBig;
            }
            else if (props.type === ButtonStyle.medium) {
                buttonStyle = mainStyles.buttonMedium;
            }
            else if (props.type === ButtonStyle.small) {
                buttonStyle = mainStyles.buttonSmall;
            }

            if (props.overrideButtonColor != null) {
                buttonStyle = [buttonStyle, { backgroundColor: props.overrideButtonColor }];
            }
        }

        return buttonStyle;
    }

    let setTextStyle = () => {
        let textStyle = mainStyles.buttonText;

        if (props.overrideTextColor != null) {
            textStyle = [textStyle, { color: props.overrideTextColor }];
        }

        return textStyle;
    }

    let showButton = () => {
        let buttonStyle = setButtonStyle();
        let textStyle = setTextStyle();

        return (
            <TouchableOpacity style={buttonStyle} onPress={() => onPress()}>
                <View style={mainStyles.nestedButtonView}>
                    {showIcon()}
                    <Text style={textStyle}>{props.text}</Text>
                </View>
            </TouchableOpacity>
        );
    }


    return (
        <View>
            {showButton()}
        </View>
    );
}
