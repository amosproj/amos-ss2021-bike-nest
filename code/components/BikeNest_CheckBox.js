import React, { useState } from 'react';
import { mainStyles } from "../styles/MainStyles";
import { Text, View } from 'react-native';
import CheckBox from '@react-native-community/checkbox';

export default function BikeNest_CheckBox(props) {
    const [toggleCheckBox, setToggleCheckBox] = useState(props.initialValue);

    function onPressTxt() {
        if (props.onPressText != null) {
            props.onPressText();
        }
    }

    return (
        <View style={mainStyles.checkBoxContainer}>
            <Text style={mainStyles.checkboxText}
                onPress={() => onPressTxt()}>{props.toggleText}</Text>
            <CheckBox style={mainStyles.checkbox}
                onValueChange={(newValue) => setToggleCheckBox(newValue)}
                value={toggleCheckBox}
            />
        </View>
    );
}