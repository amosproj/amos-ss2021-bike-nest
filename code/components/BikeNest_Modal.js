import React, { useState } from 'react';
import { Pressable, Text, View } from 'react-native';
import { Alert, Modal } from 'react-native';
import { mainStyles } from "../styles/MainStyles";

export default function BikeNest_Modal(props) {
    //const [modalVisible, setModalVisible] = useState(props.isVisible);

    function onPress() {
        if (props.onPress != null) {
            props.onPress();
        } else {
            console.error("onPress is null");
        }
    }

    function onRequestClose() {
        if (props.onRequestClose != null) {
            props.onRequestClose();
        } else {
            console.error("onRequestClose is null");
        }
    }

    return (
        <Modal
            animationType="slide"
            transparent={true}
            visible={props.isVisible}
            onRequestClose={() => onRequestClose()}
        >
            <View style={mainStyles.modalContainer}>
                <View style={mainStyles.modalContentContainer}>
                    <Text style={mainStyles.stdText}>{props.modalText}</Text>
                    <Pressable style={mainStyles.buttonSmall} onPress={() => onPress()}>
                        <Text style={mainStyles.buttonText}>OK</Text>
                    </Pressable>
                </View>
            </View>
        </Modal >);
}
