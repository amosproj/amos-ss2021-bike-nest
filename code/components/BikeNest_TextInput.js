import React from 'react';
import { mainStyles } from "../styles/MainStyles";
import { TextInput } from 'react-native';


export default function BikeNest_TextInput(props) {

    function onChangeText() {
        if (props.onChangeText != null) {
            props.onChangeText();
        }
    }

    return (
        <TextInput
            placeholder={props.placeholder}
            secureTextEntry={props.secureTextEntry}
            style={mainStyles.inputField}
            onChangeText={() => onChangeText()}
            value={props.value}
        />
    );
}