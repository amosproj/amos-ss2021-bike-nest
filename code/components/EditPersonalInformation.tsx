import React, {useState} from 'react';
import { StyleSheet, Text, View, TextInput } from 'react-native';
import {UserDataService} from "../services/UserData";

export default function EditPersonInformation() {
    let userdata: UserDataService = new UserDataService();
    const [firstName, setFirstName] = useState(userdata.GetFirstName())
    const [lastName, setLastName] = useState(userdata.GetLastName())
    const [email, setEmail] = useState(userdata.GetEmail())
    const [oldPassword, setOldPassword] = useState("")
    const [newPassword1, setNewPassword1] = useState("")
    const [newPassword2, setNewPassword2] = useState("")
    const [isEditing, setIsEditing] = useState(false)

    let saveData = () => {
        if(newPassword1.length > 0 && newPassword2.length > 0){
            if(newPassword1 !== newPassword2)
                return "New passwords don't match!";
            if(!userdata.UpdatePassword(oldPassword, newPassword1)){
                return "Error updating the password. Maybe your old password is incorrect.";
            }
        }
        let finalResult = [userdata.UpdateFirstName(firstName),
            userdata.UpdateLastName(lastName), userdata.UpdateEmail(email)].every(x => x);

        setIsEditing(false);
        return finalResult ? "Successful updated all User Data!" : "Error updating User Data!";
    };

    let cancel = () => {
        setFirstName(userdata.GetFirstName());
        setLastName(userdata.GetLastName());
        setEmail(userdata.GetEmail());
        setIsEditing(false);
    }

    let EditOrShow = (text, setter) =>
        isEditing ? <TextInput style={styles.Input} onChangeText={setter} value={text}/> : <Text>{text}</Text>

    return (
        <View>
            <View style={styles.table}>
                <View style={styles.row}>
                    <View style={styles.cell}><Text>Last Name:</Text></View>
                    <View style={styles.cell}>{EditOrShow(lastName, setLastName)}</View>
                </View>
                <View style={styles.row}>
                    <View style={styles.cell}><Text>First Name:</Text></View>
                    <View style={styles.cell}>{EditOrShow(firstName, setFirstName)}</View>
                </View>
                <View style={styles.row}>
                    <View style={styles.cell}><Text>Email Address:</Text></View>
                    <View style={styles.cell}>{EditOrShow(email,setEmail)}</View>
                </View>
            </View>
            {isEditing ?
            <View>
                <button onClick={() => cancel()}>Cancel</button>
                <button onClick={() => saveData()}>Save</button>
            </View>
                :
                <button onClick={() => setIsEditing(true)}>Edit</button>
            }

        </View>
    );
}

const styles = StyleSheet.create({
    table: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    },
    row: {
        flex: 1,
        alignSelf: 'stretch',
        flexDirection: 'row'
    },
    cell: {
        flex: 1,
        alignSelf: 'stretch'
    },
    input: {
        flex: 1,
        alignSelf: 'stretch',
        borderWidth: 5,
    }
});
