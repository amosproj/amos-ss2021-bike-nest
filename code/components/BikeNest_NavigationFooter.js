import React from 'react';
import { View } from 'react-native';
import { mainStyles } from "../styles/MainStyles";
import BikeNest_Button, { ButtonStyle } from './BikeNest_Button';
import { useNavigation } from '@react-navigation/native';

export default function BikeNest_NavigationFooter() {
    const navigation = useNavigation();

    return (
        <View style={mainStyles.navigationBar}>
            <BikeNest_Button
                type={ButtonStyle.navigationBar}
                iconPath={require("../assets/icons/user.png")}
                text="Profil"
                onPress={() => navigation.navigate("Profile")}
            />
            <BikeNest_Button
                type={ButtonStyle.navigationBar}
                iconPath={require("../assets/icons/unlock.png")}
                text="(un)lock"
                onPress={() => navigation.navigate("LockManagement")}
            />
            <BikeNest_Button
                type={ButtonStyle.navigationBar}
                iconPath={require("../assets/icons/PlusCircle.png")}
                onPress={() => navigation.navigate("EditPersonalInformation")}
            />
            <BikeNest_Button
                type={ButtonStyle.navigationBar}
                iconPath={require("../assets/icons/map-pin.png")}
                text="Navigation"
                onPress={() => navigation.navigate("FindBikeNest")}
            />
            <BikeNest_Button
                type={ButtonStyle.navigationBar}
                iconPath={require("../assets/icons/clock.png")}
                text="Verlauf"
                onPress={() => navigation.navigate("History")}
            />
        </View>
    );
}


