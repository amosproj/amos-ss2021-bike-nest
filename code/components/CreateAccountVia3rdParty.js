import React from 'react';
import { Alert, View } from 'react-native';
import colors from "../styles/Colors";
import { useNavigation } from '@react-navigation/native';
import BikeNest_Button, { ButtonStyle } from './BikeNest_Button';
import { mainStyles } from "../styles/MainStyles";
import * as WebBrowser from 'expo-web-browser';
import * as Google from 'expo-auth-session/providers/google';
import * as SecureStore from 'expo-secure-store';

export function CreateAccountVia3rdParty() {
    const navigation = useNavigation();
    const MY_SECURE_AUTH_STATE_KEY = 'MySecureAuthStateKey';

    async function getValueFor(key) {
        let result = await SecureStore.getItemAsync(key);
        if (result) {
            alert("🔐 Here's your value 🔐 \n" + result);
        } else {
            alert('No values stored under that key.');
        }
    }
    // const signInWithGoogle = () => {
    //     signInWithGoogleAsync()
    // }

    // async function signInWithGoogleAsync() {
    //     try {
    //         const result = await Google.logInAsync({
    //             behavior: 'web',
    //             //iosClientId: IOS_CLIENT_ID,
    //             //TODO:Hide client id 
    //             androidClientId: "1066777740971-p4f0ja4gl7sc8h1ingn9lo2gorc3qjts.apps.googleusercontent.com",
    //             scopes: ['profile', 'email'],
    //         });

    //         if (result.type === 'success') {
    //             return result.accessToken;
    //         } else {
    //             return { cancelled: true };
    //         }
    //     } catch (e) {
    //         return { error: true };
    //     }
    // }

    const [request, response, promptAsync] = Google.useAuthRequest({
        expoClientId: '1066777740971-sa5hjlbu6ucmequmlumjo5mresuh11n4.apps.googleusercontent.com',
        iosClientId: 'GOOGLE_GUID.apps.googleusercontent.com',
        androidClientId: "1066777740971-p4f0ja4gl7sc8h1ingn9lo2gorc3qjts.apps.googleusercontent.com",
        webClientId: '1066777740971-sa5hjlbu6ucmequmlumjo5mresuh11n4.apps.googleusercontent.com',
    });

    React.useEffect(() => {
        console.log("test");
        if (response?.type === 'success') {
            const { authentication } = response;
            console.log("success");

            const auth = response.params;
            const storageValue = JSON.stringify(auth);

            if (Platform.OS !== 'web') {
                // Securely store the auth on your device
                SecureStore.setItemAsync(MY_SECURE_AUTH_STATE_KEY, storageValue);
            }
            //let value = SecureStore.getItemAsync(MY_SECURE_AUTH_STATE_KEY)
            getValueFor(MY_SECURE_AUTH_STATE_KEY);
            navigation.navigate("FindBikeNest");
        }
        else {
            console.log(response?.type);
        }
    }, [response]);

    return (
        <View style={[mainStyles.container, { backgroundColor: 'transparent' }]}>
            <BikeNest_Button
                type={ButtonStyle.big}
                text="MIT FACEBOOK ANMELDEN"
                iconPath={require("../assets/facebook_button_logo.png")}
                overrideButtonColor={colors.Facebook_Blue}
                onPress={() => Alert.alert("Nice!",
                    "Du hast erfolgreich einen Account erstellt.",
                    [
                        { text: "OK", onPress: () => navigation.navigate("FindBikeNest") }
                    ]
                )}
            />
            <BikeNest_Button
                type={ButtonStyle.big}
                text="MIT GOOGLE ANMELDEN"
                iconPath={require('../assets/google_button_logo.png')}
                overrideButtonColor={colors.UI_BaseGrey_5}
                overrideTextColor={colors.UI_BaseGrey_0}
                onPress={() => promptAsync()}
            />
        </View>
    );
}