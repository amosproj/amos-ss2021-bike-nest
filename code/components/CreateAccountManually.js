import React, { useState } from 'react';
import { Pressable, Text, View, TextInput, StyleSheet } from 'react-native';
import { UserDataService } from "../services/UserData";
import { styles } from "./EditPersonalInformation/styles";
import { useNavigation } from '@react-navigation/native';
import { Alert, Modal } from 'react-native';
import { mainStyles } from "../styles/MainStyles";
import CheckBox from '@react-native-community/checkbox';

export function CreateAccountManually() {
    let userdata = new UserDataService();
    const [firstName, setFirstName] = useState("");
    const [lastName, setLastName] = useState("");
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");
    const [modalVisible, setModalVisible] = useState(false);
    const [modalText, setModalText] = useState("");
    const [isValidInput, setIsValidInput] = useState(false);
    const [toggleCheckBox, setToggleCheckBox] = useState(false);
    const navigation = useNavigation();

    //TODO: real Validation + Checkbox
    let validateInput = () => {
        let isValid = newEmail() && (firstName.length > 0) && (lastName.length > 0) && (email.length > 0) && (password.length > 0);
        setIsValidInput(isValid);

        if (!newEmail()) {
            setModalText("Ein Account mit der Email " + email + " existiert bereits.");
        } else {
            if (isValid)
                setModalText("Hurra! Dein Account wurde erfolgreich eingerichtet");
            else
                setModalText("Oops da ist etwas schief gelaufen. FÃ¼lle bitte alle Felder aus.")
        }
        setModalVisible(true);
    }

    let onModalPress = () => {
        if (isValidInput) {
            setModalVisible(false);
            navigation.navigate("FindBikeNest");
        }
        else
            setModalVisible(false);
    }

    //TODO: real check
    let newEmail = () => {
        let isNewEmail = true;

        if (email === '1')
            isNewEmail = false;

        return isNewEmail;
    }

    //TODO: Replace Modal (not working in web)
    return (
        <View style={styles.container}>
            <Modal
                animationType="slide"
                transparent={true}
                visible={modalVisible}
                onRequestClose={() => {
                    setModalVisible(!modalVisible);
                }}
            >
                <View style={stylesTest.modalContainer}>
                    <View style={stylesTest.modalContentContainer}>
                        <Text style={mainStyles.stdText}>{modalText}</Text>
                        <Pressable style={mainStyles.buttonSmall} onPress={() => onModalPress()}>
                            <Text style={styles.buttonText}>OK</Text>
                        </Pressable>
                    </View>
                </View>
            </Modal>
            <TextInput
                placeholder='Vorname'
                style={styles.inputField}
                onChangeText={setFirstName}
                value={firstName}
            />
            <TextInput
                placeholder='Name'
                style={styles.inputField}
                onChangeText={setLastName}
                value={lastName}
            />
            <TextInput
                placeholder='Email'
                style={styles.inputField}
                onChangeText={setEmail}
                value={email}
            />
            <TextInput
                placeholder="Passwort"
                secureTextEntry={true}
                style={styles.inputField}
                onChangeText={setPassword}
                value={password}
            />
            <View style={styles.checkBoxContainer}>
                <Text style={styles.checkboxText}
                    onPress={() => Alert.alert("Lorem ipsum")}>Datenschutzrichtlinien gelesen</Text>
                <CheckBox style={styles.checkbox}
                    onValueChange={(newValue) => setToggleCheckBox(newValue)}
                    value={toggleCheckBox}
                />
            </View>
            <Pressable style={mainStyles.buttonBig} onPress={() => validateInput()}>
                <Text style={styles.buttonText}>Los geht's</Text>
            </Pressable>
        </View>
    );
}

const stylesTest = StyleSheet.create({
    modalContentContainer: {
        width: 288,
        height: 184,
        borderRadius: 15,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#fff',
        elevation: 3,
        padding: 18,
    },
    modalContainer: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#00000090',

    }
});