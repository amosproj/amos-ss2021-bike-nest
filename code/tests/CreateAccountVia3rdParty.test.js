import React from 'react';
import renderer from 'react-test-renderer';
import { CreateAccountVia3rdParty } from '../components/CreateAccountVia3rdParty';

it('renders correctly', () => {
  const tree = renderer
    .create(<CreateAccountVia3rdParty />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});