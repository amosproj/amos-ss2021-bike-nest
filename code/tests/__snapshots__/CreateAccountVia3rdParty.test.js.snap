// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4267B2",
        "borderRadius": 38,
        "elevation": 3,
        "height": 55,
        "justifyContent": "center",
        "margin": 9,
        "opacity": 1,
        "padding": 18,
        "width": 328,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "height": 32,
            "left": 18,
            "position": "absolute",
            "top": 0,
            "width": 32,
          }
        }
      >
        <Image
          source={1}
        />
      </View>
      <Text
        style={
          Object {
            "color": "#F2F2F2",
            "fontSize": 14,
            "height": "100%",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        MIT FACEBOOK ANMELDEN
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 38,
        "elevation": 3,
        "height": 55,
        "justifyContent": "center",
        "margin": 9,
        "opacity": 1,
        "padding": 18,
        "width": 328,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "height": 32,
            "left": 18,
            "position": "absolute",
            "top": 0,
            "width": 32,
          }
        }
      >
        <Image
          source={1}
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            },
            Object {
              "color": "#000000",
            },
          ]
        }
      >
        MIT GOOGLE ANMELDEN
      </Text>
    </View>
  </View>
</View>
`;
